src/operation/get-parameters-as-json-schema.ts

```diff
@@ -179,19 +179,31 @@ export default function getParametersAsJSONSchema(
       return false;
     }
 
-    const components: Partial<ComponentsObject> = {};
+    const components: Partial<ComponentsObject> = {
+      /**
+       * Initializing an empty `components[componentType] = {}` object within the `forEach` below
+       * is incredibly slow because each of these component types has a wide variety of shapes. So
+       * in order to not have TS compilation times that takes literally **seconds** because of a
+       * single line we're instead opting to prefill this object with some empty placeholders that
+       * we'll later remove if they didn't get used.
+       *
+       * Obviously not ideal but I'd rather have a couple lines of boilerplate nonsense than having
+       * to wait a noticeably frustrating amoutn of time for TS Intellisense to reload itself after
+       * you save a line in any file.
+       */
+      examples: {},
+      schemas: {},
+      responses: {},
+      parameters: {},
+      requestBodies: {},
+      headers: {},
+      securitySchemes: {},
+      links: {},
+      callbacks: {},
+    };
 
     Object.keys(api.components).forEach((componentType: keyof ComponentsObject) => {
       if (typeof api.components[componentType] === 'object' && !Array.isArray(api.components[componentType])) {
-        /**
-         * Typescript is INCREDIBLY SLOW parsing this one line. I think it's because of the large
-         * variety of types that that object could represent but I can't yet think of a way to get
-         * around that.
-         *
-         * @todo
-         */
-        components[componentType] = {};
```

Comment:
This has been plauging this repo, and any implementation of `oas` since we migrated this library to TS. I frankly got sick of having to wait 2+ seconds for TS Intellisense to reload it self, and not being able to get anywhere with trying to improve the TS perf on this that I decided to blow it up into the above refactor.

I don't love it but I _do_ love not having to literally wait forever for Intellisense to reload itself after saving any change to any TS file in this repo.

Reply:
Out of interest how did you debug that this line was causing such a slowdown?

Reply:
I don't remember exactly how @Dashron and I narrowed this down originally but I believed it involved some sort of tsc disgnosis run and looking through the call tree.

Reply:
I did this: https://github.com/microsoft/TypeScript/wiki/Performance#performance-tracing it gives you a performance file that can be viewed in chromes debugger, and made it very clear that that one line was the slowest. commenting it out proved the issue.